src/WebJobs.Script/Config/TransmissionStatusHandler.cs

```diff
@@ -0,0 +1,148 @@
+﻿//Copyright(c).NET Foundation.All rights reserved.
+//Licensed under the MIT License. See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.Json.Serialization;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace Microsoft.Azure.WebJobs.Script.Config
+{
+    /// <summary>
+    /// Handler is used to track AI ingestion service response by subscribing to the transmission event.
+    /// The notification includes information about the response, and also the telemetry items which were part of the transmission.
+    /// </summary>
+    internal class TransmissionStatusHandler : IDisposable
+    {
+        private static readonly DiagnosticListener _source = new (ScriptConstants.HostDiagnosticSourcePrefix + "ApplicationInsights");
+
+        internal static void Handler(object sender, TransmissionStatusEventArgs args)
+        {
+            if (args == null)
+            {
+                _source.Write("TransmissionStatus", "Unable to parse transmission status response, args is null");
+                return;
+            }
+
+            // Always log if the response is non-success or if the feature flag is enabled
+            if (args.Response?.StatusCode != 200 || _source.IsEnabled(ScriptConstants.HostDiagnosticSourceDebugEventNamePrefix))
+            {
+                _source.Write("TransmissionStatus", FormattedLog(sender, args));
+            }
+        }
+
+        internal static string FormattedLog(object sender, TransmissionStatusEventArgs args)
+        {
+            var transmission = sender as Transmission;
+            var items = transmission?.TelemetryItems.GroupBy(n => n.GetEnvelopeName())
+                        .Select(n => new TelemetryItem
+                        {
+                            Type = n.Key,
+                            Count = n.Count()
+                        });
+
+            IngestionServiceResponse response = null;
+            if (!string.IsNullOrWhiteSpace(args?.Response?.Content))
+            {
+                response = JsonSerializer.Deserialize(
+                    args.Response.Content, IngestionServiceResponseContext.Default.IngestionServiceResponse);
+            }
+
+            string topErrorMessage = null;
+            int? topStatusCode = null;
+            if (response?.Errors != null && response.Errors.Length > 0)
+            {
+                topErrorMessage = response.Errors[0].Message;
+                topStatusCode = response?.Errors[0].StatusCode;
```

Comment:
a) because response? has already been checked in the if-statement and b) for consistency with line 58(either they should both have ? or not not have it 

```suggestion
                topStatusCode = response.Errors[0].StatusCode;
```